plugins/modules/user_role_workflow_manager.py

```diff
@@ -3021,11 +3021,16 @@ def get_diff_deleted(self, config):
                 self.status = "failed"
                 return self
 
-            self.msg = (
-                "Please provide a valid 'username' or 'email' for user deletion, or "
-                "The Catalyst Center user '{0}' does not have the necessary permissions "
-                "to delete a user through the API.".format(self.payload.get("dnac_username"))
-            )
+            if config.get("username") is not None:
```

Comment:
If the messages are same, then we can make it below.


if config.get("username") is not None:
    user_identifier = self.want.get("username")
else:
    user_identifier = self.want.get("email")

self.msg = (
    "The specified user '{0}' does not exist in Cisco Catalyst Center. "
    "Please provide a valid 'username' or 'email' for user deletion.".format(user_identifier)
)

self.log(self.msg, "ERROR")
self.status = "failed"
return self

Reply:
Updated the code Madhan.